### PR TITLE
feat(query): add string function: split and split_part

### LINKED ISSUE #13286: Feature: support string function split & split_part
**Summary**

support  split  & split_part

https://docs.snowflake.com/en/sql-reference/functions/split
https://docs.snowflake.com/en/sql-reference/functions/split_part

### PR DESCRIPTION
**Pull request by TCeason**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

- [x] split
- [x] split_part

- Close #13286

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/13303)

### CODE REVIEW

File: tests/sqllogictests/suites/query/02_function/02_0067_function_string_split [hunk 1]
```diff
@@ -0,0 +1,52 @@
+query T
+select split('127.0.0.1', '.');
```
Comment (BohuTANG) [inline tests/sqllogictests/suites/query/02_function/02_0067_function_string_split:2]: Need a case:
`select split('|', '|');`

Comment (TCeason) [reply tests/sqllogictests/suites/query/02_function/02_0067_function_string_split:2]: Already add more test.

**Review by BohuTANG — APPROVED** [review]
LGTM

File: src/query/functions/src/scalars/string.rs [hunk 2]
```diff
@@ -757,6 +758,73 @@ pub fn register(registry: &mut FunctionRegistry) {
...
+
+    registry
+        .register_passthrough_nullable_2_arg::<StringType, StringType, ArrayType<StringType>, _, _>(
+            "split",
+            |_, _, _| FunctionDomain::Full,
+            vectorize_with_builder_2_arg::<StringType, StringType, ArrayType<StringType>>(
+                |str, sep, output, ctx| match String::from_utf8(str.to_vec()) {
+                    Ok(s) => match String::from_utf8(sep.to_vec()) {
+                        Ok(sep) => {
+                            let res: Vec<&str> = s.split(&sep).collect();
```
Comment (sundy-li) [inline src/query/functions/src/scalars/string.rs:770]: collect could be avoided

File: src/query/functions/src/scalars/string.rs [hunk 3]
```diff
@@ -757,6 +758,73 @@ pub fn register(registry: &mut FunctionRegistry) {
...
+                            let len = res.len();
+                            let mut builder = StringColumnBuilder::with_capacity(len, len);
+                            for i in res {
+                                builder.put_slice(i.as_bytes());
+                                builder.commit_row();
+                            }
+                            let column = builder.build();
+                            output.builder.append_column(&column);
```
Comment (sundy-li) [inline src/query/functions/src/scalars/string.rs:778]: this is inefficient.   You can use the inner builder and manage the offset manually.

File: src/query/functions/src/scalars/string.rs [hunk 4]
```diff
@@ -757,6 +758,73 @@ pub fn register(registry: &mut FunctionRegistry) {
...
+
+    registry
+        .register_passthrough_nullable_3_arg::<StringType, StringType, NumberType<i64>, StringType, _, _>(
+            "split_part",
+            |_, _, _, _| FunctionDomain::Full,
+            vectorize_with_builder_3_arg::<StringType, StringType, NumberType<i64>, StringType>(
+                |str, sep, part, output, ctx| match String::from_utf8(str.to_vec()) {
+                    Ok(s) => match String::from_utf8(sep.to_vec()) {
+                        Ok(sep) => {
+                            let split: Vec<&str> = s.split(&sep).collect();
```
Comment (sundy-li) [inline src/query/functions/src/scalars/string.rs:802]: .

**Pull request merged**